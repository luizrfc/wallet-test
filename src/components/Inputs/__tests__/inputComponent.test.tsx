import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Input from '../InputComponent';
import { object } from 'yup';
import { FieldError } from 'react-hook-form';
import renderer from 'react-test-renderer';

describe('Input component', () => {
    const field = {
        name: 'name',
        ref: object,
        value: '',
        onChange: jest.fn(),
        onBlur: jest.fn(),
    };

    it('renders correctly with label', () => {
        const { getByTestId, getByText } = render(
            <Input
                field={field}
                label="Name"
                placeholder="Enter your name"
            />,
        );
        const label = getByTestId('label-test-id');
        const input = getByTestId('input-test-id');

        expect(label).toBeTruthy();
        expect(getByText('Name')).toBeTruthy();
        expect(input.props.placeholder).toBe('Enter your name');
    });

    it('renders correctly without label', () => {
        const { getByTestId } = render(
            <Input
                field={field}
                placeholder="Enter your name"
            />,
        );
        const input = getByTestId('input-test-id');
        expect(input.props.placeholder).toBe('Enter your name');
    });

    it('calls onChange and onBlur functions when typing and blurring', () => {
        const { getByTestId } = render(
            <Input
                field={field}
                label="Name"
                placeholder="Enter your name"
            />,
        );
        const input = getByTestId('input-test-id');

        fireEvent.changeText(input, 'John');
        expect(field.onChange).toHaveBeenCalledWith('John');

        fireEvent(input, 'blur');
        expect(field.onBlur).toHaveBeenCalled();
    });

    it('render input with helper to show errors message', () => {
        const messageError: FieldError = {
            type: 'error',
            message: "Error message"
        }
        const { getByTestId, getByText } = render(
            <Input
                field={field}
                label="Name"
                placeholder="Enter your name"
                helperText={messageError}
            />,
        );
        const input = getByTestId('input-test-id');
        const helper = getByTestId('helper-test-id');

        fireEvent.changeText(input, 'John');
        expect(field.onChange).toHaveBeenCalledWith('John');
        expect(helper).toBeTruthy();
        expect(getByText("Error message")).toBeTruthy();
    });

    it('render Input component correctly', () => {
        const component = renderer.create(<Input
            field={field}
            label="Name"
            placeholder="Enter your name"
        />).toJSON();
        expect(component).toMatchSnapshot();
    })
});
