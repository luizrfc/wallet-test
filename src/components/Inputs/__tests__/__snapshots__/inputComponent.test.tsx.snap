// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component render Input component correctly 1`] = `
<View
  style={
    {
      "paddingHorizontal": 2,
      "paddingVertical": 2,
    }
  }
>
  <View>
    <View
      style={
        {
          "marginBottom": 8,
        }
      }
      testID="label-test-id"
    >
      <Text
        style={
          [
            {
              "color": "#BBB",
              "fontFamily": "PTSans-CaptionRegular",
            },
            {
              "fontSize": 14,
              "lineHeight": 16,
            },
          ]
        }
        testID="text-id"
      >
        Name
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderColor": "#FFF",
          "borderRadius": 8,
          "borderWidth": 1,
          "elevation": 2,
          "flexDirection": "row",
          "paddingHorizontal": 8,
          "paddingVertical": 2,
          "shadowColor": "#000",
          "shadowRadius": 2,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        {
          "width": "100%",
        }
      }
    >
      <TextInput
        onBlur={
          [MockFunction] {
            "calls": [
              [],
            ],
            "results": [
              {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        onChangeText={
          [MockFunction] {
            "calls": [
              [
                "John",
              ],
              [
                "John",
              ],
            ],
            "results": [
              {
                "type": "return",
                "value": undefined,
              },
              {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        placeholder="Enter your name"
        placeholderTextColor="#BBB"
        testID="input-test-id"
        value=""
      />
    </View>
  </View>
</View>
`;
